### PR TITLE
Replace document and window with function getters

### PR DESCRIPTION
**Pull request by veekeys**
After #99 SSR build was broken again. This time with document in FilterToggle.
Decided to replace all document/window with functions, where we check if those are defined (this is [recommended ](https://nextjs.org/docs/migrating/from-create-react-app#safely-accessing-web-apis) to do in SSR applications).

### CODE REVIEW

File: src/core/utils/common.tsx [hunk 1]
```diff
@@ -60,3 +60,19 @@ export const getContainer = (containerId: string, ownerDocument = document) => {
   }
   return container;
 };
+
```
Comment (bentleyvk) [inline src/core/utils/common.tsx:63]: Maybe change to `getDocument`
image

File: src/core/utils/hooks/useIntersection.tsx [hunk 2]
```diff
@@ -24,7 +25,7 @@ export const useIntersection = (
 
   const setRef = React.useCallback(
     (node: HTMLElement | null) => {
-      if (!window.IntersectionObserver) {
+      if (!getWindow()?.IntersectionObserver) {
```
[the remark this answers is not in the document]
Comment (veekeys) [reply src/core/utils/hooks/useIntersection.tsx:28]: Where? 
Would it run with that if above?

File: src/core/utils/common.tsx (hunk 1 above)
Comment (veekeys) [reply src/core/utils/common.tsx:66]: added as default parameter value

File: src/core/utils/common.tsx [hunk 3]
```diff
@@ -51,12 +51,31 @@ export const getUserColor = (emailOrName: string) => {
...
     ownerDocument.body.appendChild(container);
   }
   return container;
 };
+
+/**
+ * Get document if it is defined.
+ * Used to support SSR/SSG applications.
+ */
+export const getDocument = () => {
```
Comment (mayank99) [inline src/core/utils/common.tsx:71]: Wonder if this should take an optional Element parameter and use that to return `element.ownerDocument`.

File: src/core/Table/filters/FilterToggle.tsx [hunk 4]
```diff
@@ -50,7 +51,7 @@ export const FilterToggle = <T extends Record<string, unknown>>(
           placement='bottom'
           visible={isVisible}
           onClickOutside={close}
-          appendTo={ownerDocument.body}
+          appendTo={ownerDocument?.body}
```
Comment (mayank99) [inline src/core/Table/filters/FilterToggle.tsx:54]: I would have just gone with `() => ownerDocument.body` 😄

File: src/core/utils/common.tsx (hunk 3 above)
Comment (veekeys) [reply src/core/utils/common.tsx:71]: I would like not to include any other logic in here..ownerDocument stuff is different.

File: src/core/Table/filters/FilterToggle.tsx (hunk 4 above)
Comment (veekeys) [reply src/core/Table/filters/FilterToggle.tsx:54]: huh? How that would have helped

Comment (mayank99) [reply src/core/Table/filters/FilterToggle.tsx:54]: Huh.
image

Well ok 🤷‍♀️

Comment (veekeys) [reply src/core/Table/filters/FilterToggle.tsx:54]: Yes, it was in FilterToggle... Initially, wanted to fix only that, but...
Decided to just replace all places not to bother about stuff anymore.

**Review by mayank99 — APPROVED** [review]

**Review by mayank99 — APPROVED** [review]

**Review by mayank99 — APPROVED** [review]

**Pull request merged**